practice/rain.py

```diff
@@ -0,0 +1,130 @@
+"""Takes data from rain gauge at Sunnyside and prints a summary of the data: date with the most rain &
+year with most rain.
+"""
+# setup:
+filename = 'sunnyside.txt'
+
+
+def each_line_of_file(filename):
+    """Opens sunnyside.txt and reads it line by line."""
+    with open(filename, newline='') as f:
+        lines = f.readlines()
+
+    return lines
+
+
+def split_lines(lines):
+    r"""Strips and splits the lines into lists.
+
+    >>> split_lines(['1 \n', '2\n', '3\n', '4\n', '5\n', '6 \n', '7 \n', '8 \n', '9 \n', '10 \n', '11 \n', 'PARTY! \n', 'TIME!'])
+    [['PARTY!']]
+    """
+    listed_lines = [x.strip().split('\n') for x in lines]
+    listed_lines_reduced = listed_lines[11:-1]
+    line_list = []
+    for listed_lines_split in listed_lines_reduced:
+        line_list += [y.split() for y in listed_lines_split]
+
+    return line_list
+
+
+def remove_junk(line_list):
+    """Cleans up the list and sets daily rain amount to an int.
+
+    >>> remove_junk([['test', '1', '2', '3']])
+    [['test', 1]]
```

Comment:
Part of the functionality of this function is removing `-` entries. That should be doctested.

An argument could me made that those are two separate steps and each should be it's own function.